### ISSUE TITLE
redundant objects are not caught by lints

### ISSUE THREAD

**Issue by h1alexbel** [objectionary/lints#376]
Consider this EO program:

```eo
[] > check-array
  list > created
    * 1 1 2
  is-empty.
    created 
```

Here, `created` is [redundant](https://www.yegor256.com/2015/09/01/redundant-variables-are-evil.html) object, it was introduced to be used only _once_. It can be inlined like this:

```eo
[] > check-array
  is-empty.
    list
      * 1 1 2
```

Of course this situation should not be reported:

```eo
[] > uses-in-loop
  list > created
    * 1 2 3
  list
    * 1 2 3
  .eachi > @
    [item index] >>
      withi.
        created
        index
        item
```

no defect here, because, `created` was introduced in order to avoid re-creation of array inside the loop (`.eachi`)

**Comment by h1alexbel** [comment]
@yegor256 @maxonfjvipon WDYT?

**Comment by maxonfjvipon** [comment]
@h1alexbel sometimes in order to keep code readable we can move some big parts of it to separated attributes (like we move some functionality to separated functions in Java), so I'm not sure how we can strictly describe which object should be inlined and which are not. I believe the programmer should decide it

**Comment by h1alexbel** [comment]
@maxonfjvipon agree, but thats why I suggest to make `warning` severity of this lint, not an error. @yegor256 WDYT?

**Comment by yegor256** [comment]
@maxonfjvipon I'm leaning towards this rule. Such objects are obviously a redundancy. If they exist, the code is begging for refactoring. This lint may highlight this situation.

**Comment by h1alexbel** [comment]
@yegor256 should we add this issue into backlog?

**Comment by yegor256** [comment]
@h1alexbel yes, let's do it

**Comment by 0pdd** [comment]
@h1alexbel the puzzle [#592](https://github.com/objectionary/lints/issues/592) is still not solved.

**Comment by 0pdd** [comment]
@h1alexbel the only puzzle [#592](https://github.com/objectionary/lints/issues/592) is solved here.
